src/Dscp/Educator/Secret/Real/Functions.hs

```diff
@@ -51,11 +53,13 @@ genStore = do
     return EducatorSecret{..}
 
 -- | Read store under given path.
-readStore :: (MonadIO m, MonadCatch m) => FilePath -> PassPhrase -> m EducatorSecret
+readStore
+    :: (MonadIO m, MonadCatch m, MonadLogging m)
```

Comment:
Why did you add `MonadLogging` here, if you didn't add any actual logging? :thinking:

Reply:
`ensureModeIs` can report upon fixing file mode

Reply:
Ok, sorry, missed it.